### PR TITLE
General algorithm improvements and additions

### PR DESCRIPTION
**Pull request by RealEnder**
Based on #28 
Tested successfully with captures on http://wpa-sec.stanev.org

### CODE REVIEW

**Comment by ruiaraujo** [comment]
Thanks for this. I will try to review it ASAP. It will get a review until Sunday at most!

**Comment by RealEnder** [comment]
Now I plan to sync algorithms and tests from routerkeygenAndroid, but it will take some time.

File: src/main.cpp [hunk 1]
```diff
@@ -29,83 +29,89 @@
...
-            mac = "";
-            if ( !options.value("q", false).toBool() )
-                std::cout << QObject::tr("Invalid MAC. It will not be used.").toUtf8().data() << std::endl;
+    if (argc > 1) {
+        QCoreApplication app(argc, argv);
+        QCmdLineParser parser;
+        parser.addOption("-s", QCmdLineArgument::StoreValue, QObject::tr("SSID"), "--ssid", "network_name");
+        parser.addOption("-m", QCmdLineArgument::StoreValue, QObject::tr("MAC address"), "--mac", "mac_address");
+        parser.addOption("-v", QCmdLineArgument::StoreTrue, QObject::tr("Version"), "--version");
+        parser.addOption("-k", QCmdLineArgument::StoreTrue, QObject::tr("Print keygen in form kgname:candidate"), "--kg");
```
Comment (ruiaraujo) [inline src/main.cpp]: The kgname in all those classes is only for the CLI tool?

Comment (RealEnder) [inline src/main.cpp]: Yes, couldn't think of simpler solution, without RTTI, which would be overkill.

**Comment by ruiaraujo** [comment]
Squashed and merged.
I think a lot of this can be synced to RKAndroid but it will require some effort.

Thanks for the help.

**Pull request merged**